sqlmesh/core/model/decorator.py

```diff
@@ -123,7 +127,7 @@ def model(
         for key in ("pre_statements", "post_statements"):
             statements = common_kwargs.get(key)
             if statements:
-                common_kwargs[key] = [exp.maybe_parse(s, dialect=dialect) for s in statements]
+                common_kwargs[key] = [parse_one(s, dialect=dialect) for s in statements]
```

Comment:
Can `statements` ever contain actual expression instances instead of strings? This will crash if this happens. Perhaps that's why we were using `exp.maybe_parse` here in the first place?

Reply:
right, I didn't consider this. I've adjusted it to allow the use of actual expression instances